src/connector.ts

```diff
@@ -53,30 +55,41 @@ export declare interface DriverOptions {
 class CloudSQLInstanceMap extends Map {
   async loadInstance({
     ipType,
+    authType,
     instanceConnectionName,
     sqlAdminFetcher,
   }: {
     ipType: IpAddressTypes;
+    authType: AuthTypes;
     instanceConnectionName: string;
     sqlAdminFetcher: SQLAdminFetcher;
   }): Promise<void> {
     // in case an instance to that connection name has already
     // been setup there's no need to set it up again
     if (this.has(instanceConnectionName)) {
       const instance = this.get(instanceConnectionName);
-      if (instance.ipType !== ipType) {
+      if (instance.ipType && instance.ipType !== ipType) {
```

Comment:
```suggestion
      if (instance.ipType !== ipType) {
```

I don't see why the extra check is necessary, can we remove it?

Reply:
nvm! I just run the tests and I see now that we need that extra check to see if `ipType` is defined 😬